Helper/AdyenOrderPayment.php

```diff
@@ -214,7 +222,9 @@ public function createAdyenOrderPayment(Order $order, Notification $notification
     {
         $adyenOrderPayment = null;
         $payment = $order->getPayment();
-        $amount = $this->adyenDataHelper->originalAmount($notification->getAmountValue(), $order->getBaseCurrencyCode());
+        $chargedCurrencyCode = $this->config->getChargedCurrency() === "display"
+            ? $order->getOrderCurrencyCode() : $order->getBaseCurrencyCode();
+        $amount = $this->adyenDataHelper->originalAmount($notification->getAmountValue(), $chargedCurrencyCode);
```

Comment:
@dejankar would you mind doing quick check for this testing your scenario?

```suggestion
        $amount = $this->adyenDataHelper->originalAmount($notification->getAmountValue(), $notification->getAmountCurrency());
```

Reply:
> @dejankar would you mind doing quick check for this testing your scenario?

Hi Angel, I don't mind and I already tested your suggestion even before and can confirm that's working as expected. 

Just keep me informed if anything changes in the way how things are passed back from the webhooks.

Glad that the solution is there and I will make sure to add the change in our code base, so we follow the plugin development. 

Best Regards,
Dejan

Reply:
> > @dejankar would you mind doing quick check for this testing your scenario?
> 
> Hi Angel, I don't mind and I already tested your suggestion even before and can confirm that's working as expected when capturing payments.
> 
> However when refunding from the Adyen platform the amount is wrong, and always tries to refund the base amount instead of the correct amount.
> 
> I'm checking if this is related with this code or not.
> 
> Dejan

Seems that issue is not related with those changes and it's another issue. 

So when order is placed with one of KWD, OMR, BHD as a display currency and then I try to refund from Adyen platform the full amount. 

This is the response in M2: 

Adyen Refund Successfully completed

IPN "Refunded". Refund issued by merchant. Registered notification about refunded amount of $15.90. Transaction ID: "". Credit Memo has not been created. Please create offline Credit Memo.

Adyen HTTP Notification(s):
eventCode: REFUND
pspReference: JR6PFJ8VBGNG5S82
paymentMethod: visa
success: true

Reply:
Hi @acampos1916 , 
Nothing in the logs that clarifies it. 

The notification seems correct and data inside is correct: 
[2022-03-16 09:11:43] AdyenLoggerTest.ADYEN_NOTIFICATION: The content of the notification item is: {"amount":{"currency":"KWD","value":15900},"eventCode":"REFUND","eventDate":"2022-03-16T10:10:38+01:00","merchantAccountCode":"Lens_Me","merchantReference":"1000612602","originalReference":"JXVHHCW6PP4WHD82","paymentMethod":"visa","pspReference":"JR6PFJ8VBGNG5S82","reason":"","success":"true"} {"is_exception":false} []
[2022-03-16 09:11:43] AdyenLoggerTest.ADYEN_NOTIFICATION: The result is accepted {"is_exception":false} []

I guess it's the same issue how the amount is being formated when sent back from Adyen webhooks, but will debug later on and get back when I have more info. 

It's not a general bug as i tried the same with our base currency and a refund was successfully created.

Best Regards,
Dejan